TravelTrack-API.Tests/UserServiceAsyncTests.cs

```diff
@@ -0,0 +1,318 @@
+
+        [TestCategory("GetAll"), TestCategory("Successful Functionality"), TestMethod]
+        public async Task UserService_Get_ReturnsListOfAllUsers()
+        {
+            // Arrange
+            TravelTrackContext _ctx = NewContext();
+            IUserService userService = new UserService(_ctx, _mapper);
+            int actualCount = _ctx.Users.Count();
+
+            // Act
+            List<UserDto> usersResult = await userService.GetAllAsync();
+
+            // Assert 
+            Assert.IsInstanceOfType(usersResult, typeof(List<UserDto>));
+            Assert.AreEqual(3, usersResult.Count()); // 3 is the number of users seeded in TravelTrackContext class
+            Assert.AreEqual("jmoran@ceiamerica.com", usersResult[0].Username);
+            Assert.AreEqual("dummyuser@dummy.dum", usersResult[1].Username);
+            Assert.AreEqual("fakeuser@fakey.fake", usersResult[2].Username);
+        }
+
+        // Tests for Get()
+
+        [TestCategory("Get"), TestCategory("Successful Functionality"), TestMethod]
+        public async Task UserService_Get_ReturnsCorrectUser()
+        {
+            // Arrange
+            TravelTrackContext _ctx = NewContext();
+            IUserService userService = new UserService(_ctx, _mapper);
+            string username = "jmoran@ceiamerica.com";
+
+            // Act
+            UserDto userResult = await userService.GetAsync(username);
+
+            // Assert 
+            Assert.AreEqual(username, userResult.Username);
+        }
+
+        [TestCategory("Get"), TestCategory("Not Found"), TestMethod]
+        public async Task UserService_Get_ThrowsNotFoundException()
+        {
+            // Arrange
+            TravelTrackContext _ctx = NewContext();
+            IUserService userService = new UserService(_ctx, _mapper);
+            string username = "nonexistinguser@test.test";
+
+
+            // Act
+            try
+            {
+                UserDto userResult = await userService.GetAsync(username);
+
+                // Assert
+                Assert.Fail();
+            }
+            catch (HttpResponseException e)
+            {
+                //Assert
+                Assert.AreEqual(HttpStatusCode.NotFound, e.Response.StatusCode);
+                Assert.AreEqual($"Username Not Found", e.Response.ReasonPhrase);
+            }
+        }
+
+
+        // Tests for Add()
+
+        [TestCategory("Add"), TestCategory("Successful Functionality"), TestMethod]
+        public async Task UserService_Add_ReturnsSuccessfullyAddedUser()
+        {
+            // Arrange
+            TravelTrackContext _ctx = NewContext();
+            IUserService userService = new UserService(_ctx, _mapper);
+            int priorUserCount = _ctx.Users.Count();
+            UserDto newUser = new UserDto
+            {
+                Username = "newuser@test.test",
+                Password = "P@ssw0rd",
+                FirstName = "New", 
+                LastName = "User"
```

Comment:
A good way to find problems in a test is to use data that doesn't fit what you're expecting. For all the test data I use i prefix it with "Unit Test" So my username would be "Unit Test Username". You'd be surprised the problems you can find when you pass in unexpected data. you'll find that you rely on a well formed structure in your code.